components/event-bus/internal/knative/subscription/controller/subscription/controller.go

```diff
@@ -1,60 +1,51 @@
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
 
-var log = logf.Log.WithName("subscription-controller")
+	eventbusscheme "github.com/kyma-project/kyma/components/event-bus/client/generated/clientset/internalclientset/scheme"
+	eventbusclient "github.com/kyma-project/kyma/components/event-bus/client/generated/injection/client"
+	eventactivationinformersv1alpha1 "github.com/kyma-project/kyma/components/event-bus/client/generated/injection/informers/applicationconnector/v1alpha1/eventactivation"
+	subscriptioninformersv1alpha1 "github.com/kyma-project/kyma/components/event-bus/client/generated/injection/informers/eventing/v1alpha1/subscription"
+	"github.com/kyma-project/kyma/components/event-bus/internal/knative/subscription/opts"
+	"github.com/kyma-project/kyma/components/event-bus/internal/knative/util"
+)
 
 const (
+	// reconcilerName is the name of the reconciler
+	reconcilerName = "Subscriptions"
+
 	// controllerAgentName is the string used by this controller to identify
 	// itself when creating events.
 	controllerAgentName = "subscription-controller"
 )
 
-// ProvideController instantiates a reconciler which reconciles Kyma Subscriptions.
-func ProvideController(mgr manager.Manager, opts *opts.Options) error {
-
-	// init the knative lib
-	knativeLib, err := knative.NewKnativeLib()
-	if err != nil {
-		log.Error(err, "Failed to get Knative library")
-		return err
-	}
-
-	// Setup a new controller to Reconcile Kyma Subscription.
-	r := &reconciler{
-		recorder:   mgr.GetRecorder(controllerAgentName),
-		opts:       opts,
-		knativeLib: knativeLib,
-		time:       util.NewDefaultCurrentTime(),
-	}
-	c, err := controller.New(controllerAgentName, mgr, controller.Options{
-		Reconciler: r,
-	})
-	if err != nil {
-		log.Error(err, "Unable to create controller")
-		return err
-	}
+func init() {
+	// Add sources types to the default Kubernetes Scheme so Events can be
+	// logged for sources types.
+	eventbusscheme.AddToScheme(scheme.Scheme)
+}
 
-	// Watch Subscriptions.
-	err = c.Watch(&source.Kind{
-		Type: &eventingv1alpha1.Subscription{},
-	}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		log.Error(err, "Unable to watch Subscription")
-		return err
+// NewController returns a new controller that reconciles EventActivation objects.
+func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+	subscriptionInformer := subscriptioninformersv1alpha1.Get(ctx)
+	eventActivationInformer := eventactivationinformersv1alpha1.Get(ctx)
+
+	r := &Reconciler{
+		Base:                  reconciler.NewBase(ctx, controllerAgentName, cmw),
+		subscriptionLister:    subscriptionInformer.Lister(),
+		eventActivationLister: eventActivationInformer.Lister(),
+		kymaEventingClient:    eventbusclient.Get(ctx).EventingV1alpha1(),
+		opts:                  opts.DefaultOptions(),
+		time:                  util.NewDefaultCurrentTime(),
 	}
+	impl := controller.NewImpl(r, r.Logger, reconcilerName)
 
```

Comment:
We need to watch for changes to (Kyma) Subscription objects, otherwise the reconciliation won't get triggered.

```golang
// set event handlers
subscriptionInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
```